patching/src/main/java/org/jboss/as/patching/metadata/PatchMerger.java

```diff
@@ -0,0 +1,512 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.patching.metadata;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jboss.as.patching.Constants;
+import org.jboss.as.patching.IoUtils;
+import org.jboss.as.patching.PatchingException;
+import org.jboss.as.patching.ZipUtils;
+import org.jboss.as.patching.logging.PatchLogger;
+import org.jboss.as.patching.metadata.Patch.PatchType;
+import org.jboss.as.patching.runner.PatchUtils;
+
+/**
+ *
+ * @author Alexey Loubyansky
+ */
+public class PatchMerger {
+
+    private static final String DIRECTORY_PREFIX = "wildfly-patch-";
```

Comment:
There's no fix needed for this PR, but should this use some kind of configurable property so it will work with EAP as well?